spec/unit/middleware/timeout_spec.rb

```diff
@@ -0,0 +1,177 @@
+require 'spec_helper'
+require 'qless/job'
+require 'qless/middleware/timeout'
+
+module Qless
+  module Middleware
+    ::Rspec.describe Timeout do
+      class JobClass
+        attr_accessor :perform
+
+        def around_perform_call_counter
+          @around_perform_call_counter ||= 0
+        end
+
+        def around_perform(job)
+          @around_perform_call_counter = (@around_perform_call_counter || 0) + 1
+        end
+      end
+
+      let(:kernel_class) { class_double(Kernel) }
+
+      let(:job) {
+        instance_double(Qless::Job)
+      }
+
+      def make_worker(timeout_class, timeout_seconds, kernel_class)
+        Class.new(JobClass) do
+          include Qless::Middleware::Timeout.new(timeout_class: timeout_class,
+              kernel_class: kernel_class) { timeout_seconds }
+        end.new
+      end
+
+      before {
+        allow(job).to receive(:reconnect_to_redis)
+        allow(job).to receive(:klass_name)
+        allow(job).to receive(:fail)
+        allow(kernel_class).to receive(:exit!)
```

Comment:
Apparently, in current version of RSpec the order cannot be changed. It annoys me :-( but stimulates RSpec migration